lamindb/_finish.py

```diff
@@ -208,23 +208,28 @@ def clean_r_notebook_html(file_path: Path) -> tuple[str | None, Path]:
 
 
 def check_filepath_recently_saved(filepath: Path, is_retry: bool) -> bool:
-    recently_saved_time = 2.5 if not is_retry else 10
+    base_time = 2.5 if not is_retry else 15
+    file_size_mb = filepath.stat().st_size / (1024 * 1024)
+    # Scale wait time: add 5 seconds per 10MB, cap at 60 seconds
+    size_adjustment = min(file_size_mb / 10 * 5, 45)
+    recently_saved_time = base_time + size_adjustment
+
+    wait_message = "waiting for notebook to be saved"
```

Comment:
We need to wrap this into the `get_save_notebook_message()` message because the 2 notebook cleaning functions depend on it (they'll clean up the R notebook html and the Jupyter JSON).